src/contracts/Protocol/ClaimController.sol

```diff
@@ -19,21 +59,24 @@ contract ClaimController is Ownable, IClaimController, OnlyApi {
         claimLimits[questionId] = claimLimit;
     }
 
-    function claim(uint256 questionId) public onlyOwner {
+    function claim(address user, uint256 questionId) public onlyApi {
         if (claims[questionId].length >= claimLimits[questionId]) revert ClaimLimitReached();
+        if (answers[questionId][user].author == user) revert AlreadyClaimed();
 
-        claims[questionId].push(_msgSender());
-        Answer memory _answer = Answer({state: STATE.CLAIMED, author: _msgSender(), answerURL: "", scoringMetaDataURI: "", finalGrade: 0});
-        answers[questionId][_msgSender()] = _answer;
+        claims[questionId].push(user);
+        Answer memory _answer = Answer({state: STATE.CLAIMED, author: user, answerURL: "", scoringMetaDataURI: "", finalGrade: 0});
+        answers[questionId][user] = _answer;
     }
 
-    function answer(uint256 questionId, string calldata answerURL) public onlyOwner {
-        if (answers[questionId][_msgSender()].state != STATE.CLAIMED) revert NeedClaimToAnswer();
-        answers[questionId][_msgSender()].answerURL = answerURL;
+    function answer(
+        address user,
+        uint256 questionId,
+        string calldata answerURL
+    ) public onlyOwner {
+        if (answers[questionId][user].state != STATE.CLAIMED) revert NeedClaimToAnswer();
+        answers[questionId][user].answerURL = answerURL;
```

Comment:
Hmm are we not updating more properties in this function call or are we implementing separate function calls to update other states like finalGrade, scoringMetaDataURI?
or maybe answer isn't the best name for this IDK?

Reply:
Not too sure yet, currently this function is still untouched (also inaccessible because its still onlyOwner haha), once we get to work on figuring out the details of answering I think this will solve itself.

Reply:
Like the other ones can you add a TODO for this as well